src/ngraph/frontend/onnx_import/op/shape.cpp

```diff
@@ -32,11 +32,20 @@ namespace ngraph
             {
                 NodeVector shape(const Node& node)
                 {
-                    auto data = node.get_ng_inputs().at(0);
-                    auto data_shape = data->get_shape();
-
-                    return {std::make_shared<default_opset::Constant>(
-                        ngraph::element::i64, Shape{data_shape.size()}, data_shape)};
+                    const auto data = node.get_ng_inputs().at(0);
+                    const auto data_ps = data->get_output_partial_shape(0);
```

Comment:
Please use `data_shape` or full `data_partial_shape` if it makes more sense in this case.
```suggestion
                    const auto data_shape = data->get_output_partial_shape(0);
```